pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -244,3 +433,103 @@ func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context
+	}
+
+	if err := r.Get(ctx, infraMachineKey, infraMachine); err != nil {
+		return nil, nil, fmt.Errorf("failed to get CAPI infrastructure machine: %w", err)
+	}
+
+	return infraCluster, infraMachine, nil
+}
+
+// capiInfraMachineIsEqual checks whether the provided CAPI infra machines are equal.
+func capiInfraMachineIsEqual(platform configv1.PlatformType, infraMachine1, infraMachine2 client.Object) (bool, error) {
+	switch platform {
+	case configv1.AWSPlatformType:
+		typedInfraMachine1, ok := infraMachine1.(*capav1beta2.AWSMachine)
+		if !ok {
+			return false, errAssertingCAPIAWSMachine
+		}
+
+		typedinfraMachine2, ok := infraMachine2.(*capav1beta2.AWSMachine)
+		if !ok {
+			return false, errAssertingCAPIAWSMachine
+		}
+
+		return reflect.DeepEqual(typedInfraMachine1.Spec, typedinfraMachine2.Spec) && util.ObjectMetaIsEqual(typedInfraMachine1.ObjectMeta, typedinfraMachine2.ObjectMeta), nil
+	case configv1.PowerVSPlatformType:
+		typedInfraMachine1, ok := infraMachine1.(*capibmv1.IBMPowerVSMachine)
+		if !ok {
+			return false, errAssertingCAPIIBMPowerVSMachine
+		}
+
+		typedinfraMachine2, ok := infraMachine2.(*capibmv1.IBMPowerVSMachine)
+		if !ok {
+			return false, errAssertingCAPIIBMPowerVSMachine
+		}
+
+		return reflect.DeepEqual(typedInfraMachine1.Spec, typedinfraMachine2.Spec) && util.ObjectMetaIsEqual(typedInfraMachine1.ObjectMeta, typedinfraMachine2.ObjectMeta), nil
+	default:
+		return false, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
+	}
+}
+
+// applySynchronizedConditionWithPatch updates the synchronized condition
+// using a server side apply patch. We do this to force ownership of the
+// 'Synchronized' condition and 'SynchronizedGeneration'.
+func (r *MachineSyncReconciler) applySynchronizedConditionWithPatch(ctx context.Context, mapiMachine *machinev1beta1.Machine, status corev1.ConditionStatus, reason, message string, generation *int64) error {
+	var severity machinev1beta1.ConditionSeverity
+	if status == corev1.ConditionTrue {
+		severity = machinev1beta1.ConditionSeverityNone
+	} else {
+		severity = machinev1beta1.ConditionSeverityError
+	}
+
+	conditionAc := machinev1applyconfigs.Condition().
+		WithType(consts.SynchronizedCondition).
+		WithStatus(status).
+		WithReason(reason).
+		WithMessage(message).
+		WithSeverity(severity)
+
+	util.SetLastTransitionTime(consts.SynchronizedCondition, mapiMachine.Status.Conditions, conditionAc)
+
+	statusAc := machinev1applyconfigs.MachineStatus().
+		WithConditions(conditionAc)
+
+	if status == corev1.ConditionTrue && generation != nil {
+		statusAc = statusAc.WithSynchronizedGeneration(*generation)
+	}
+
+	msAc := machinev1applyconfigs.Machine(mapiMachine.GetName(), mapiMachine.GetNamespace()).
+		WithStatus(statusAc)
+
+	if err := r.Status().Patch(ctx, mapiMachine, util.ApplyConfigPatch(msAc), client.ForceOwnership, client.FieldOwner(controllerName+"-SyncronizedCondition")); err != nil {
```

Comment:
Please check if there are expectations on the names of the field owners, they may supposed to be lowercase or not include hyphens, I'm not sure but we should check the requirements
```suggestion
	if err := r.Status().Patch(ctx, mapiMachine, util.ApplyConfigPatch(msAc), client.ForceOwnership, client.FieldOwner(controllerName+"-SynchronizedCondition")); err != nil {
```

Reply:
Yes I checked. It dows work.

```
apiVersion: machine.openshift.io/v1beta1
kind: Machine
metadata:
  creationTimestamp: "2025-02-25T08:48:58Z"
  generateName: foo
  generation: 1
  managedFields:
  - apiVersion: machine.openshift.io/v1beta1
    fieldsType: FieldsV1
    manager: MachineSyncController-SynchronizedCondition
    operation: Apply
    subresource: status
    time: "2025-02-25T08:48:58Z"
...
```